### ISSUE TITLE
Cannot access ports in devel.yaml

### ISSUE THREAD

**Issue by Yajo** [Tecnativa/docker-odoo-base#120]
It seems that #117 raised yet another issue: cannot access ports in `devel.yaml`.

I opened https://github.com/moby/moby/issues/36174 to track it, docker-side.

For now, the fastest workaround would be to **set `internal: false` in `devel.yaml` and then `docker-compose down` and `docker-compose up -d` again**.

I'll investigate a better workaround when I have time.

**Comment by Yajo** [comment]
I'm thinking that maybe for devel.yaml we should just default to a normal network. Installing packages and so on could be more common than restoring a database in localhost. In such cases, one should be warned about temporarily changing  the internal flag.

Another option would be bundling a small tcp proxy container that is the only one with external access and redirects ports to other containers in the stack.

**Comment by pedrobaeza** [comment]
Questions for this:

- Removing internal, are we losing Mailhog?
- What about wdb?

**Comment by Yajo** [comment]
Removing internal in `devel.yaml` (setting it to a normal network), what you loose is the isolation from the www, useful mainly to restore production databases and have no fear on outside world pollution.

By adding the proxy instead of that, we lose no features isolation. It's just that `devel.yaml` becomes less obvious.

In both cases, mailhog and wdb will be still available.

In any case it's just a workaround for moby/moby#36174 that I hope gets fixed soon.

**Comment by pedrobaeza** [comment]
Let's see if they say something. For now, let's use the workaround.

**Comment by Yajo** [comment]
Upstream is still deciding if this is a bug, a missing feature or a design decision. Should we start working in a downstream fix?

**Comment by pedrobaeza** [comment]
Can we?

**Comment by Yajo** [comment]
tl;dr: yes.

We'd need a proxy container that is connected to both networks. Pretty similar to what is done in `test.yaml`, where [the traefik proxy is connected to both internal and external networks](https://github.com/Tecnativa/docker-odoo-base#global-inverse-proxy).

Odoo, mailhog, wdb, and all containers in the stack would be in a internal-only network. The proxy would be in that one and in a normal network too, and would be the door to access internal services. This way we have whitelisted isolation.

The same principle would be applied to whitelist some external services in `test.yaml`, such as google fonts for reports, etc.

**Comment by pedrobaeza** [comment]
OK, let's work in a prototype then.

**Comment by Yajo** [comment]
Fixed in #126
